src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java

```diff
@@ -1,11 +1,90 @@
 package frc.team2412.robot.subsystem;
 
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.TalonFXFeedbackDevice;
 import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
-
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
 public class ShooterSubsystem extends SubsystemBase {
-    public ShooterSubsystem(WPI_TalonFX flywheelMotor1, WPI_TalonFX flywheelMotor2, WPI_TalonFX turretMotor, WPI_TalonFX hoodMotor) {
+    private final WPI_TalonFX flywheelMotor1;
+    private final WPI_TalonFX flywheelMotor2;
+    private final WPI_TalonFX turretMotor;
+    private final WPI_TalonFX hoodMotor;
+
+    public static final double MAX_FORWARD = 0.1;
+    public static final double MAX_REVERSE = -0.1;
+    public static final double STOP_MOTOR = 0;
+    public static final double DEGREES_TO_ENCODER_TICKS = 2048 / 360; // 2048 ticks per 360 degrees
+    public static final double MIN_TURRET_ANGLE = 0;
+    public static final double MAX_TURRET_ANGLE = 180; // Placeholder, maybe need to update
+
+    public ShooterSubsystem(WPI_TalonFX flywheelMotor1, WPI_TalonFX flywheelMotor2, WPI_TalonFX turretMotor,
+            WPI_TalonFX hoodMotor) {
+        this.flywheelMotor1 = flywheelMotor1;
+        this.flywheelMotor2 = flywheelMotor2;
+        this.flywheelMotor2.setInverted(true);
+        this.turretMotor = turretMotor;
+        this.hoodMotor = hoodMotor;
```

Comment:
Please configure the rest of the motor options too:
Set max current limit to reasonable value (likely ~30amp for flywheels, ~10amp for others)
Set the flywheels to free run
Set the turret & hood to brake mode
Set them inverted (or not) as appropriate
Likely set defaults for everything else

Reply:
For the hood & turret, consider setting soft limits on the positions so the motor can internally help keep the turret & hood from moving beyond its physical limits.

Reply:
I talked with Tristan, everything is just 1:1 belts, so we don't need to compensate for gear ratio or invert motors (except for the second flywheel motor).